apps/aecore/src/aec_tx_sign.erl

```diff
@@ -18,15 +20,35 @@ verify(failed_tx) ->
 verify(_Tx) ->
     ok.
 
-%% TODO This is meant to be the deterministic canonical serialization.
-serialize(SignedTx = #signed_tx{data = Tx, signatures = [_]}) when
-      is_tuple(Tx) ->
-    term_to_binary(SignedTx).
-
-%% TODO This is meant to be the deserialization of the deterministic
-%% canonical serialization.
-deserialize(B) ->
-    case binary_to_term(B) of
-        #signed_tx{signatures = [_]} = SignedTx ->
-            SignedTx
-    end.
+-spec serialize(#signed_tx{}) -> map().
+serialize(#signed_tx{data = Tx, signatures = Sigs})  when is_tuple(Tx) ->
+    {Data, Type} =
+      case Tx of
+          #coinbase_tx{} ->
+              {aec_coinbase_tx:serialize(Tx), <<"coinbase">>}
+      end,
```

Comment:
@velzevur please refactor that so we don't populate the code with cases of transaction

Lets let only tx behaviour have this case